src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -55,31 +54,38 @@ public void AddBodyParameter(string key, string value)
             {
                 _bodyParameters[key] = value;
             }
-        }
+        }        
 
         internal void AddHeader(string key, string value)
         {
             _headers[key] = value;
         }
 
+        internal IReadOnlyDictionary<string, string> GetBodyParameters()
+        {
+            return new ReadOnlyDictionary<string, string>(_bodyParameters);
+        }
+
         public async Task<InstanceDiscoveryResponse> DiscoverAadInstanceAsync(Uri endPoint, RequestContext requestContext)
         {
             return await ExecuteRequestAsync<InstanceDiscoveryResponse>(endPoint, HttpMethod.Get, requestContext)
                        .ConfigureAwait(false);
         }
 
-        public async Task<MsalTokenResponse> GetTokenAsync(Uri endPoint, RequestContext requestContext)
+        internal async Task<MsalTokenResponse> GetTokenAsync(Uri endPoint, RequestContext requestContext)
         {
             return await ExecuteRequestAsync<MsalTokenResponse>(endPoint, HttpMethod.Post, requestContext).ConfigureAwait(false);
         }
 
         internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, RequestContext requestContext, bool expectErrorsOn200OK = false)
         {
-            bool addCorrelationId = requestContext != null && !string.IsNullOrEmpty(requestContext.Logger.CorrelationId.ToString());
+            CheckParamsNotNull(endPoint, requestContext);
```

Comment:
will the rc ever be null? this is created in MSAL, so not really anything the developer can control. We should always have some value for this, correct? seems like a check that doesn't need to happen here, or we need to ensure we always have a value

Reply:
It's fair point. I've removed this. We guarantee these are not-null.